src/shogun/multiclass/KNN.cpp

```diff
@@ -123,33 +121,66 @@ SGMatrix<index_t> KNN::nearest_neighbors()
 	distance->precompute_lhs();
 	distance->precompute_rhs();
 
-	//for each test example
-	for (auto i : SG_PROGRESS(range(n)))
+	switch (m_knn_solver)
 	{
-		COMPUTATION_CONTROLLERS
-		//lhs idx 0..num train examples-1 (i.e., all train examples) and rhs idx i
-		distances_lhs(dists,0,m_train_labels.vlen-1,i);
-
-		//fill in an array with 0..num train examples-1
-		for (int32_t j=0; j<m_train_labels.vlen; j++)
-			train_idxs[j]=j;
-
-		//sort the distance vector between test example i and all train examples
-		Math::qsort_index(dists.vector, train_idxs.vector, m_train_labels.vlen);
-
-#ifdef DEBUG_KNN
-		io::print("\nQuick sort query {}\n", i);
-		for (int32_t j=0; j<m_k; j++)
-			io::print("{} ", train_idxs[j]);
-		io::print("\n");
-#endif
-
-		//fill in the output the indices of the nearest neighbors
-		for (int32_t j=0; j<m_k; j++)
-			NN(j,i) = train_idxs[j];
+	case KNN_BRUTE:
+	{
+		//for each test example
+		for (auto i : SG_PROGRESS(range(n)))
+		{
+			COMPUTATION_CONTROLLERS
+			//lhs idx 0..num train examples-1 (i.e., all train examples) and rhs idx i
+			distances_lhs(dists,0,m_train_labels.vlen-1,i);
+
+			//fill in an array with 0..num train examples-1
+			train_idxs.range_fill(0);
+
+			std::pair<float64_t, index_t> pairt[m_train_labels.vlen]; 
+  
+			// Storing the respective array 
+			// elements in pairs. 
+			for (int j = 0; j < m_train_labels.vlen; j++)  
+			{ 
+				pairt[j].first = dists[j]; 
+				pairt[j].second = train_idxs[j]; 
+			} 
+
+			// Sorting the pair array. 
+			sort(pairt, pairt + m_train_labels.vlen); 
```

Comment:
so if this compiles, this means that there is a `using namespace std` lingering somehwere, which is bad. Could you check where that is? Need to follow the includes ... best use grep or something

Reply:
`using namespace std;` is found in 22 files in the `src/shogun` directory out of which 11 instances are in the `src/shogun/io` directory.
Should I remove them and refactor the code accordingly?

Reply:
It is ok for using the namespace in .cpp files. In header files we need to avoid it.
Could you find out which changes you need to make such that this file doesnt compile anymore?